### ISSUE TITLE
User defined KeyWords are not recognized by the Intercactive Console 

### ISSUE THREAD

**Issue by Alan-Carr** [robocorp/robotframework-lsp#713]
Entering a user defined KeyWord in the interactive console produces Error "No keyword with came "xxxx" found.

My user defined KeyWord executes fine within a test

Do I have to somehow enable ?

**Comment by fabioz** [comment]
You need to import it.

i.e.: You need to type in the interactive console something as:

```
*** Settings ***
Library    my_library.py
Resource    my_resource.robot
```

to add the related keywords to your namespace.

**Comment by Alan-Carr** [comment]
I have my .robot file open within VSC this contains test cases and some keywords that I have defined based on keywords from libraries which are defined in Settings.

Prior to your reply I would load my "settings" section via "Load in Interactive console"  do I need to add a resource section within settings which is the name of my .robot file ?

I tried by typing -:
*** Settings ***
Resource    x.robot (name of my .robot file already open in VSC

**Comment by fabioz** [comment]
If you want to load all the keywords from a resource (even if it's the one open in VSC), then yes, you need to add a `Resource    x.robot` there (note that it'll load the contents from disk in this case, so, if you have edits that are unsaved you need to save the file first).

Usually for the current file it's better to define the keywords by clicking over the `Load in interactive console` over each keyword you want from the open file (but there's currently no way to load all except by creating a `Resource` import in a `Settings` section).

**Comment by Alan-Carr** [comment]
I think I need to define full path to my .robot file

**Comment by fabioz** [comment]
Humm, that's not it, the problem is that anything imported in a `Resource` section can't have `Setup` nor `Test Case` sections.

So, for this to work you need to move the keywords you want to a `.resource` (and make the test use that file as well as the interactive console).

Or you need to import keyword by keyword manually...

**Comment by fabioz** [comment]
As a note, what you could do is just copy/paste the whole `*** Keywords ***` section (as well as the needed `*** Settings ***`)... this is probably the easiest thing to do right now ;)

**Comment by Alan-Carr** [comment]
Hi Fabio

Apologizes as I still very new to Robot Framework and ever newer to VSC

1/ I copied from the keywords section one of my user defined keywords and my settings section in an attempt to get my custom keywords running in the interactive console.

***@***.***

***@***.***

When I try to run the keyword from the interactive prompt I get

***@***.***

2/ We spoke previously about when multiple libraries have the same keyword eg Click when a user selects the keyword required from the popup the keyword doesn't include the library name at the beginning of the keyword eg Sikuli.Click xxxxx or RPA.Desktop.Click xxxx.

3/ Is there anyway to show all keywords with their associated libraries in a list like

***@***.***

Alan




Alan CARR - Senior QC Engineer
T. +441182286646 ***@***.******@***.***>

EVS Broadcast Equipment - Building B, Crowthorne House Nine Mile Ride - RG40 3GA Wokingham Berkshire - UNITED KINGDOM


Quicklinks:   Latest News<http://www.evs.com/news>   Next Events<http://www.evs.com/events>   Contacts<http://www.evs.com/contact>
Community:   Facebook<http://www.facebook.com/home.php#!/pages/EVS-Broadcast-Equipment/262186759015?ref=ts>   Twitter<http://twitter.com/EVSEquipment>    LinkedIn<https://www.linkedin.com/company/evs-broadcast-equipment>  Youtube<http://www.youtube.com/user/EVSBroadcast>   Instagram<http://instagram.com/evsequipment>   MyEVS<http://www.evs.com/myevs>   RSS<http://www.evs.com/rss/news>

<http://www.evs.com/>[cid:signature_uk-move(2)_6cb7465f-ba2f-4893-a077-582100954528.jpg]<https://evs.com/evs-uk-moving>

www.evs.com<http://www.evs.com>


EVS Broadcast Equipment is a commercial name which is used for all legal entities of our group<https://evs.com/sites/default/files/evs_broadcast_equipment_addresses.pdf>.

At EVS we give utmost importance to your data's privacy. If you wish to know more on how we manage your data and on your rights, please consult our Privacy Statement<https://evs.com/en/privacystatement>.

This message may contain confidential information and is intended solely for the use of the addressee. Please do not use or publish the contained information without the agreement of the sender. If you are not the intended recipient of this message, please notify the sender and do not disclose, use, disseminate or copy this message.

Please consider the environment before printing this email.

**Comment by fabioz** [comment]
@Alan-Carr your response is garbled and hard to follow -- maybe you e-mail did this when you replied?

**Comment by Alan-Carr** [comment]
I took a look at my sent items within Outlook and all looked ok. (My reply contains text and graphics)

I thought it best if I emailed directly rather than using GitHub as my reply included a few different items.

How do you think we get around this ?

Alan


Alan CARR - Senior QC Engineer
T. +441182286646 ***@***.******@***.***>

EVS Broadcast Equipment - Building B, Crowthorne House Nine Mile Ride - RG40 3GA Wokingham Berkshire - UNITED KINGDOM


Quicklinks:   Latest News<http://www.evs.com/news>   Next Events<http://www.evs.com/events>   Contacts<http://www.evs.com/contact>
Community:   Facebook<http://www.facebook.com/home.php#!/pages/EVS-Broadcast-Equipment/262186759015?ref=ts>   Twitter<http://twitter.com/EVSEquipment>    LinkedIn<https://www.linkedin.com/company/evs-broadcast-equipment>  Youtube<http://www.youtube.com/user/EVSBroadcast>   Instagram<http://instagram.com/evsequipment>   MyEVS<http://www.evs.com/myevs>   RSS<http://www.evs.com/rss/news>

<http://www.evs.com/>[cid:signature_uk-move(2)_6cb7465f-ba2f-4893-a077-582100954528.jpg]<https://evs.com/evs-uk-moving>

www.evs.com<http://www.evs.com>


EVS Broadcast Equipment is a commercial name which is used for all legal entities of our group<https://evs.com/sites/default/files/evs_broadcast_equipment_addresses.pdf>.

At EVS we give utmost importance to your data's privacy. If you wish to know more on how we manage your data and on your rights, please consult our Privacy Statement<https://evs.com/en/privacystatement>.

This message may contain confidential information and is intended solely for the use of the addressee. Please do not use or publish the contained information without the agreement of the sender. If you are not the intended recipient of this message, please notify the sender and do not disclose, use, disseminate or copy this message.

Please consider the environment before printing this email.

**Comment by fabioz** [comment]
Sure, you can e-mail me at fabiofz at gmail dot com.
